pygen/pygen_src/riscv_asm_program_gen.py

```diff
@@ -550,10 +552,10 @@ def gen_trap_handler_section(self, hart, mode, cause, tvec,
                                                       cause.name) +
                          "{}srli x{}, x{}, {}\n".format(pkg_ins.indent, cfg.gpr[0].value,
                                                         cfg.gpr[0].value, rcs.XLEN - 1) +
-                         "{}bne x{}, x0, {}{}mode_instr_handler".format(pkg_ins.indent,
-                                                                        cfg.gpr[0].value,
-                                                                        pkg_ins.hart_prefix(hart),
-                                                                        mode))
+                         "{}bne x{}, x0, {}{}mode_intr_handler".format(pkg_ins.indent,
+                                                                       cfg.gpr[0].value,
```

Comment:
nit: if gpr[0] is a vsc type, maybe use get_val method instead?

Reply:
Hi @hodjat91 
Yes, its a vsc type. I tried without value it works in that case also.
I was assuming enum.name will return string name and enum.value returns an integer value.
It seems like python "{}".format() takes the integer value by default unless we specify enum.name.
Well, I have did the changes all over the asm class. 
Please take a look.